src/main/java/com/apicatalog/jdk8/Jdk8Compatibility.java

```diff
@@ -0,0 +1,58 @@
+package com.apicatalog.jdk8;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public interface Jdk8Compatibility {
```

Comment:
I don't like the class name nor package name.  Let's figure out something better.